src/legacy/core_plugins/kibana/ui_setting_defaults.js

```diff
@@ -91,7 +91,15 @@ export function getUiSettingDefaults() {
           'Query language used by the query bar. Kuery is an experimental new language built specifically for Kibana.',
       }),
       type: 'select',
-      options: ['lucene', 'kuery']
+      options: ['lucene', 'kuery'],
+      optionLabels: {
+        lucene: i18n.translate('kbn.advancedSettings.searchQueryLanguageLucene', {
+          defaultMessage: 'Lucene',
+        }),
+        kuery: i18n.translate('kbn.advancedSettings.searchQueryLanguageKql', {
+          defaultMessage: 'KQL',
```

Comment:
Lucene and KQL are "product names". I don't think we want to translate them

Reply:
I agree, but I like to still have them in a translatable string. I make sure they are added to the global glossary so translators know not to translate them, but what if even though it's called `KQL` in another language, they want to use their own script for it, e.g. it would be written in Katakana because it's a foreign word. So even though I expect it to be the same in most languages, I personally would still prefer having it using the translation API.